common/djangoapps/track/views/__init__.py

```diff
@@ -96,27 +96,23 @@ def server_track(request, event_type, event, page=None):
     except:
         username = "anonymous"
 
-    # define output:
-    event = {
-        "username": username,
-        "ip": _get_request_ip(request),
-        "referer": _get_request_header(request, 'HTTP_REFERER'),
-        "accept_language": _get_request_header(request, 'HTTP_ACCEPT_LANGUAGE'),
-        "event_source": "server",
-        "event_type": event_type,
-        "event": event,
-        "agent": _get_request_header(request, 'HTTP_USER_AGENT').decode('latin1'),
-        "page": page,
-        "time": datetime.datetime.utcnow(),
-        "host": _get_request_header(request, 'SERVER_NAME'),
-        "context": eventtracker.get_tracker().resolve_context(),
-    }
+    if isinstance(event, basestring) and len(event) > 0:
```

Comment:
No need for checking `len()`, just say `and event`

```suggestion
    if isinstance(event, basestring) and event:
```